pymc_experimental/distributions/mvncdf.py

```diff
@@ -22,38 +23,40 @@ def conditional_covariance(Sigma, mu, conditioned, conditioned_values):
     Sigma_12 = Sigma[target][:, conditioned]
     Sigma_11 = Sigma[target][:, target]
 
-    Sigma_cond = Sigma_11 - np.matmul(np.matmul(Sigma_12, np.linalg.inv(Sigma_22)), Sigma_21)
-    mean_cond = np.delete(mu, conditioned) + np.matmul(Sigma_12, np.linalg.inv(Sigma_22)).dot(
-        conditioned_values - mu[conditioned]
-    )
+    inv = pm.math.matrix_inverse(Sigma_22)
+    Sigma_cond = Sigma_11 - pm.math.matrix_dot(Sigma_12, inv, Sigma_21)
+    mean_cond = pm.math.matrix_dot(Sigma_12, inv, conditioned_values)
 
     return Sigma_cond, mean_cond
 
 
 class Mvncdf(Op):
     __props__ = ()
 
-    def make_node(self, upper, mu, cov):
-        upper = at.as_tensor_variable(upper)
+    def make_node(self, value, mu, cov):
+        value = at.as_tensor_variable(value)
         mu = at.as_tensor_variable(mu)
         cov = at.as_tensor_variable(cov)
 
-        return Apply(self, [x], [x.type()])
+        return Apply(self, [value], [value.type()])
 
     def perform(self, node, inputs, output_storage):
         # here do the integration
-        return scipy.stats.multivariate_normal.logcdf(upper, mu, cov)
+        value, mu, cov = inputs
+        output_storage[0] = scipy.stats.multivariate_normal.logcdf(value, mu, cov)
+
+        return output_storage
 
     def infer_shape(self, fgraph, node, i0_shapes):
-        return
+        return i0_shapes[0]
 
     def grad(self, inputs, output_grads):
-
+        value, mu, cov = inputs
         grad_ = []
         for i in range(len(mu)):
-            grad_.append(conditional_covariance(cov, mu, i, upper[i]))
+            grad_.append(conditional_covariance(cov, mu, i, value[i]))
 
-        return np.array(grad_)
+        return np.array(grad_)*output_grads[0]

```

Comment:
The returned gradients need to be symbolic Aesara expressions, not the result of Numpy computations. You might need a separate `Op` for the gradient. See an example here: https://www.pymc.io/projects/examples/en/latest/case_studies/blackbox_external_likelihood_numpy.html#blackbox_external_likelihood_numpy and https://www.pymc.io/projects/examples/en/latest/case_studies/wrapping_jax_function.html

Reply:
I think for now it might be best not to implement the gradient method. The scipy autograd library doesn't exist for multivariate normal cdf. For now I can just try and sample using MH.